pkg/kube/scaler.go

```diff
@@ -22,35 +22,40 @@ var ScaleToZero ScaleMutation = func(current int32) (expect int32) {
 	return 0
 }
 
-func NewScaler(logger log.Logger, clientset *kubernetes.Clientset, namespace string) *Scaler {
-	return &Scaler{logger: logger, clientset: clientset, namespace: namespace}
+func NewScaler(logger log.Logger, clientset *kubernetes.Clientset, namespace string) Scaler {
```

Comment:
The construction function is usually preferred to return a concreted type pointer instead of an interface, because the result may be served with multiple interfaces.